src/machines/confirmation-dialog-machine.ts

```diff
@@ -0,0 +1,182 @@
+// Source: https://xstate-catalogue.com/machines/confirmation-dialog
+import {assign, createMachine} from 'xstate'
+import {Dispatch, SetStateAction} from 'react'
+import {getAuthorizationHeader} from 'utils/auth'
+import axios from 'axios'
+import toast from 'react-hot-toast'
+
+type Member = {
+  id: number
+  name: string
+  email: string
+}
+
+export interface ConfirmationDialogMachineContext {
+  action?: () => Promise<void>
+  errorMessage?: string
+  accountId: number | undefined
+  memberToRemove: Member | undefined
+  setMembers: Dispatch<SetStateAction<any[]>>
+}
+
+export type ConfirmationDialogMachineEvent =
+  | {
+      type: 'OPEN_DIALOG'
+      action: () => Promise<void>
```

Comment:
The `action` is something that came directly from the machine starter I used ([here](https://xstate-catalogue.com/machines/confirmation-dialog)). It is not being used, so I should remove it.

I would like to take a separate pass over this machine and see if I can move the `removeTeamMember` function out of the machine and instead pass it in as the `action`... which is probably the more idiomatic way of doing things. I could use this same technique to move the Success and Error Toast actions outside of the machine. This would make the machine less complex and keep it a little more generic.

<img width="1082" alt="CleanShot 2021-04-27 at 15 34 41@2x" src="https://user-images.githubusercontent.com/694063/116309412-576e0000-a76e-11eb-84e1-6ccd0d90d522.png">

With regard to your question of whether the generic should be `void`: I'm not sure what would make sense here. It might just be that they gave it void as boilerplate and it is up to the specific implementation to give it a type that makes sense.